package/lib/src/controls/form_field.dart

```diff
@@ -38,6 +38,30 @@ TextInputType parseTextInputType(String type) {
   return TextInputType.text;
 }
 
+DatePickerMode parseDatePickerMode(String mode) {
+  switch (mode.toLowerCase()) {
+    case "day":
+      return DatePickerMode.day;
+    case "year":
+      return DatePickerMode.year;
+  }
+  return DatePickerMode.day;
+}
+
+DatePickerEntryMode parseDatePickerEntryMode(String mode) {
+  switch (mode.toLowerCase()) {
+    case "calendar":
+      return DatePickerEntryMode.calendar;
+    case "input":
+      return DatePickerEntryMode.input;
+    case "calendarOnly":
+      return DatePickerEntryMode.calendarOnly;
+    case "inputOnly":
+      return DatePickerEntryMode.inputOnly;
+  }
+  return DatePickerEntryMode.calendar;
+}
+
```

Comment:
I think we could move both `parseDatePickerMode()` and `parseDatePickerEntryMode()` to `date_picker.dart` as they are not going to be used anywhere else.